segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -178,19 +179,37 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
     }
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
+        if (writer.get() == null) {
+            log.warn("Event writer not set up yet. Ignoring the request");
```

Comment:
This is odd, why would this occur? If it shouldn't happen just make this a precondition.